sites/public/__tests__/pages/applications/contact/address.test.tsx

```diff
@@ -0,0 +1,72 @@
+import React from "react"
+import { setupServer } from "msw/lib/node"
+import { fireEvent } from "@testing-library/react"
+import { mockNextRouter, render } from "../../../testUtils"
+import ApplicationAddress from "../../../../src/pages/applications/contact/address"
+
+window.scrollTo = jest.fn()
+
+const server = setupServer()
+
+beforeAll(() => {
+  server.listen()
+  mockNextRouter()
+})
+
+afterEach(() => server.resetHandlers())
+
+afterAll(() => server.close())
+
+describe("applications pages", () => {
+  afterAll(() => {
+    jest.clearAllMocks()
+  })
+
+  describe("address step", () => {
+    it("should render form fields", () => {
+      const { getByText, getByTestId, getAllByTestId } = render(<ApplicationAddress />)
+
+      expect(
+        getByText("Now we need to know how to contact you about your application.", {
+          exact: false,
+        })
+      ).toBeInTheDocument()
+      expect(getByTestId("app-primary-phone-number")).toBeInTheDocument()
+      expect(getByTestId("app-primary-phone-number-type")).toBeInTheDocument()
+      expect(getByTestId("app-primary-no-phone")).toBeInTheDocument()
+      expect(getByTestId("app-primary-additional-phone")).toBeInTheDocument()
+      expect(getByTestId("app-primary-address-street")).toBeInTheDocument()
+      expect(getByTestId("app-primary-address-street2")).toBeInTheDocument()
+      expect(getByTestId("app-primary-address-city")).toBeInTheDocument()
+      expect(getByTestId("app-primary-address-state")).toBeInTheDocument()
+      expect(getByTestId("app-primary-address-zip")).toBeInTheDocument()
+      expect(getByTestId("app-primary-send-to-mailing")).toBeInTheDocument()
+      expect(getAllByTestId("app-primary-contact-preference")).toHaveLength(4)
+      expect(getByTestId("app-primary-work-in-region-yes")).toBeInTheDocument()
+      expect(getByTestId("app-primary-work-in-region-no")).toBeInTheDocument()
+    })
+
+    it("should require form input", async () => {
+      const { getByText, findByText } = render(<ApplicationAddress />)
+
+      fireEvent.click(getByText("Next"))
+      expect(
+        await findByText("There are errors you'll need to resolve before moving on.")
+      ).toBeInTheDocument()
+      expect(getByText("Please enter a phone number")).toBeInTheDocument()
+      expect(getByText("Please enter a phone number type")).toBeInTheDocument()
+      expect(getByText("Please enter an address")).toBeInTheDocument()
+      expect(getByText("Please select at least one option.")).toBeInTheDocument()
+      expect(getByText("Please select one of the options above.")).toBeInTheDocument()
```

Comment:
Should there also be a part of this test to resolve the errors and re-click "Next" to make sure errors are removed properly?

Reply:
Eventually that sounds great, it just wasn't a part of the current coverage and didn't want to bloat this PR

Reply:
Good call out. It's definitely not a blocker. I'll approve this PR and we can create some tickets for additional coverage